packages/netlify-cms-backend-test/src/implementation.ts

```diff
@@ -10,35 +10,65 @@ import {
   ImplementationEntry,
   AssetProxy,
   PersistOptions,
-  ImplementationMediaFile,
   User,
   Config,
   ImplementationFile,
 } from 'netlify-cms-lib-util';
+import { extname, dirname } from 'path';
 import AuthenticationPage from './AuthenticationPage';
 
-type RepoFile = { file?: { path: string }; content: string };
+type RepoFile = { path: string; content: string | AssetProxy };
 type RepoTree = { [key: string]: RepoFile | RepoTree };
 
+type UnpublishedRepoEntry = {
+  slug: string;
+  collection: string;
+  status: string;
+  diffs: {
+    id: string;
+    originalPath?: string;
+    path: string;
+    newFile: boolean;
+    status: string;
+    content: string | AssetProxy;
+  }[];
+  timestamp: string;
+};
+
 declare global {
   interface Window {
     repoFiles: RepoTree;
-    repoFilesUnpublished: ImplementationEntry[];
+    repoFilesUnpublished: { [key: string]: UnpublishedRepoEntry };
   }
 }
 
 window.repoFiles = window.repoFiles || {};
 window.repoFilesUnpublished = window.repoFilesUnpublished || [];
 
-function getFile(path: string) {
+function getFile(path: string, tree: RepoTree) {
```

Comment:
The test backend required quite a few changes to be able to handle multiple files per entry